src/cwe_checker_lib/src/utils/binary.rs

```diff
@@ -167,6 +167,30 @@ impl RuntimeMemoryImage {
         Err(anyhow!("Address is not a valid global memory address."))
     }
 
+    /// Read the contents of memory from a given address onwards until a null byte is reached.
+    pub fn read_string_until_null_terminator(&self, address: &Bitvector) -> Result<Vec<u8>, Error> {
+        let address = address.try_to_u64().unwrap();
+        for segment in self.memory_segments.iter() {
+            if address >= segment.base_address
+                && address <= segment.base_address + segment.bytes.len() as u64
+            {
+                let index = (address - segment.base_address) as usize;
+                let mut bytes: Vec<u8> = Vec::new();
+                for byte in segment.bytes[index..].iter() {
+                    if *byte == 0 {
+                        if self.is_little_endian {
+                            bytes = bytes.into_iter().rev().collect();
```

Comment:
This is wrong: The representation of strings is **not** affected by the endianess of the architecture.